Change TopBar.kt to use LargeTopAppBar

SuperDragonXD opened the pull request:
## Description
<!-- Please include a summary of the change. Please also include relevant motivation and context. -->
This PR changes TopBar.kt to use LargeTopAppBar. Alongside that, it bumps `androidx.compose.material3` to `beta01` (needed for some functionality to work, alongside some minor changes to support that version), and removes `rememberFloatingState()` due to it being unused.

## Type of change
<!-- Replace :x: with :white_check_mark: to "check" the specified bullet -->

:x: General change (non-breaking change that doesn't fit the below categories like copyediting)
:x: Bug fix (non-breaking change which fixes an issue)
:white_check_mark: New feature (non-breaking change which adds functionality)
:x: Breaking change (fix or feature that would cause existing functionality to not work as expected)

SuperDragonXD committed:
Change TopBar.kt to use LargeTopAppBar

yasandev commented:
Haven't fully looked into the code (yet) but the current implementation doesn't have the proper behavior; You cannot go back to the initial state by scrolling to the top. Although it sometimes happens in a glitchy way as you can see in the video.

![Peek 2022-09-04 15-47](https://user-images.githubusercontent.com/41836211/188310603-e3425f9a-b0cf-441c-a234-35c6ce1466f5.gif)

SuperDragonXD committed:
`PreferenceScaffold`: Change scroll behavior

Change scroll behavior to use `enterAlwaysScrollBehavior()` instead of `exitUntilCollapsedScrollBehavior()` to fix returning to expanded state issues.

SuperDragonXD commented:
Fixed that behavior (see most recent commit). Do you want me to make the title larger when the top bar is uncollapsed?

SuperDragonXD committed:
`TopBar`: Remove custom text weight & fix variable

Remove custom weight on Text() and return topBarSize variable to old value (since it is used in PreferenceSearchScaffold)
